### ISSUE TITLE
incompatibilities to Jira Server version

### ISSUE THREAD

**Issue by kfiry77** [ljay79/jira-tools#12]
Hi, 
I am trying to use the plugin with my company Jira, however when I open The application stack when tryng to load the filter.
I have viewed the console and observed the following error caught:
```
Uncaught Error: Adresse nicht verfügbar: https://jira.mycompany.com/rest/api/2/filter/my?includeFavourites=true
 at jiraApi:181 (Jira Sheet Tools:10) 66e1ffbf-99f0-4651-93b1-929a5146f080
 at getMyFilters (jiraCommon:220) (Jira Sheet Tools:10) b87812ce-8027-47e4-9cd0-ef98f9c5dcfe
```

My Company uses jira with http, however I see everywhere in the code : `'https://' +  ....`
Please assist.

**Comment by ljay79** [comment]
Applied a patch with commit 8147509762a7a712f1bcb4b0c24d2b4d23f1ce16 .
Already pushed to GAS as version 0.13.0, should be automatically updated in your google sheet within few hours.
Check the "About" from within google sheet to look out for version 0.13.0 which should allow you to use "http://" jira servers now.

**Comment by kfiry77** [comment]
Hi, Thanks for the patch. It is not stack anymore, 
However It seems that when trying to find the filter, i get message : 

> No filters available!

When I am looking at "Manage filter" section on the JIRA website, I am getting many filter, part of them mine, part of them public. 
Please advise.

BTW my JIRA server  version is  v7.3.0

**Comment by ljay79** [comment]
Hi, thanks for the feedback.

I was just researching difference for Jira Server and Jira Cloud.
Im using the Jira Cloud version only.

It seems, that the methid im using to fetch the users filters _/filter/my_ is not existing in the Jira Server version of the api.
I can implement a failover for this difference but wont be able to do it today anymore.

Can you tell me, if at least the other feature works: "Refresh Ticket Data"?

BTW: This Add-on is using the more simple Basic Auth method for authenticating with the jira rest api.
It is not recommended to use this without SSL (https) as the data (username and passwords) are transmitted unsecure over the internet.

**Comment by kfiry77** [comment]
Hi, 
I tried to use "Refresh Ticket Data", and it did not refresh even after 15 Minutes.
Seem to me it does not work.  Am i doing something wrong?
Thanks for the update and support. 

Regarding the SSL, I have already inform our IT/ALM to fix it, hope it will be end soon.

**Comment by ljay79** [comment]
Sry to hear. I guess it's an incompatibility to Jira server version. For development I had only an OnDemand version available.

I will see that I can make it compatible soon. Just a little harder to do, as I won't be able to test it during development.

Sry that the plugin isn't working for you yet.
But your feedback is valuable. Thx

**Comment by kfiry77** [comment]
Thanks for the support.
If there is anything else I can share, feel free to contact me directly.

**Comment by kfiry77** [comment]
Hi, 
I have looked and check the filters API since you noted that it does not exists on the Server version, 
I have found that the available filter is `rest/api/2/filter/favourite`, so I suggest to use this REST API as fallback when the /filter/my does not work.
Thanks and Regards,

**Comment by ljay79** [comment]
Ok, i have now added handling to differentiate between these 2 Jira versions (server vs. ondemand).
Again, i can only blindly build it as i dont have a Jira Server to test against with.

However, from the documentation, i could find only the filter retrieval method are different in both. The other methods so far should be compatible for both.

I have just released the new version (https://github.com/ljay79/jira-tools/tree/v0.14.0) to GAS.
It should be available shortly in the Google Sheet Addon Store (GAS).
Your installed add-on would update fully automatically.

When new update available (v0.14.0), please go into: _“Add-ons > Jira Sheet Tools > Settings”_
and press "SAVE", this will then trigger the new config for Server vs. OnDemand.

Note: You can check what version is currently available in your google sheet, by open _"Add-ons > Jira Sheet Tools > About"_ from within google sheet .

Do let me know, if it all works now for you.

**Comment by kfiry77** [comment]
That worked for getting the list of filter smoothly.
however, when I have selected one of the filter I am getting the following message:

> Failed to retrieve jira issues from filter with status [400]!
> Unrecognized field "properties" (Class com.atlassian.jira.rest.v2.search.SearchRequestBean), not marked as ignorable at [Source: org.apache.catalina.connector.CoyoteInputStream@63a9c341; line: 1, column: 184] (through reference chain: com.atlassian.jira.rest.v2.search.SearchRequestBean["properties"])

I have even tried to disable all field and I have got the same result.
Can you elaborate which REST API you are using to fetch the list, so I would check it on my server?

Once again, thank for the great support!

**Comment by ljay79** [comment]
{code}
https://docs.atlassian.com/jira/REST/server/#api/2/search-searchUsingSearchRequest

POST with 
(SAMPLE)

data{
	jql: 'assignee = currentUser() and resolution is empty', 
    fields: [array of columns as check in form], 
    properties : ['due'],
    maxResults: 100, 
    validateQuery: 'strict'
}
{code}

I just compared docu for the search method in both server version.
The "properties" property appears not be supported on Jira Server api.

I assume, you are using Jira Server rather then a Jira OnDemand instance.

I will shortly double check the difference and see, if i can fix it asap.

**Comment by ljay79** [comment]
Hi.

Ok so i found the issue and could solve it. At the same time found and (kinda) solve another issue too.
New release https://github.com/ljay79/jira-tools/tree/v0.15.0 is available in github but also any minute in the Google Add-on store.

PS: In case you find another issue (now or later) please create an individual issue here, so we can keep it clear and structured issue by issue also for other users later on.

Thx
(and again, the Jira Server compatibility im doing quite blind as i have only a Jira OnDemand available to test against with)

**Comment by ljay79** [comment]
Any feedback yet with the new changes?

**Comment by kfiry77** [comment]
OK, 
Tested it.
I am getting the message:
```
Failed to retrieve jira issues from filter with status [400]!
Can not construct instance of java.lang.Boolean from String value 'strict': only "true" or "false" recognized at [Source: org.apache.catalina.connector.CoyoteInputStream@25cc194d; line: 1, column: 309] (through reference chain: com.atlassian.jira.rest.v2.search.SearchRequestBean["validateQuery"])
```

**Comment by kfiry77** [comment]
Hi, 
I have tested the REST and found the following issues:
1. validateQuery requires bool while you entered 'strict' value.
2. properties field is not recognized.  

after fixing these two issues, I the Rest API works smoothly.
Thk.

**Comment by ljay79** [comment]
Thx for debugging... the "properties" attribute I removed in latest version already. The "validateQuery" I have to check, might be another difference to Jira Server.

Thx for debugging

**Comment by ljay79** [comment]
Yep, the param "validateQuery" for Jira Cloud (OnDemand) requires "strict" as the boolean false/true is deprecated, while on the Jira Server it is still true or false.

Will add a switch depending on the jira instance type.

**Comment by ljay79** [comment]
Ok, so the "properties" attribute issue should be fixed already since v0.15.0 and the "validateQuery" issue i have applied a fix in the latest version now v0.16.0 .

Pls check and let me know if there is anything else.

Thx again for the feedback - its very valuable.

**Comment by kfiry77** [comment]
Thank you very much. it's is now working perfectly.
be sure that I am going to use it, and distribute to all my colleges.
You support was valuable.
